src/core/manager.c

```diff
@@ -1389,6 +1389,29 @@ static unsigned manager_dispatch_gc_job_queue(Manager *m) {
         return n;
 }
 
+static int manager_ratelimit_check_and_queue(Unit *u, sd_event_time_handler_t cb) {
+        int r;
+
+        assert(u);
+        assert(cb);
+
+        if (ratelimit_below(&u->auto_start_stop_ratelimit))
+                return 1;
+
+        r = sd_event_add_time_relative(
+                        u->manager->event,
+                        &u->auto_start_stop_event_source,
```

Comment:
if auto_start_stop_event_source is already initialized you'll leak an event source here.

Reply:
you should either reuse the existing event source and restart it if it already elapsed (i.e. is disabled – but leave it on if it is still pending, and exit early here).

Or we should unref/disable the event source whenever it elapses from its callback handler.